fix(vector): warp groups using combined bounding box

LilSuperUser opened the pull request:
Groups of paths now warp together as a unit instead of independently.

- union all subpath bounding boxes into one frame
- normalize points and handles against shared bounds
- apply bilinear interpolation into target quad
- reset transforms after 

Closes #3551

LilSuperUser committed:
fix(vector): warp groups using combined bounding box

Groups of paths now warp together as a unit instead of independently.

- union all subpath bounding boxes into one frame
- normalize points and handles against shared bounds
- apply bilinear interpolation into target quad
- reset transforms after warp

Fixes #3551

Ayush2k02 commented on node-graph/nodes/vector/src/vector_nodes.rs:
```diff
@@ -817,20 +832,22 @@ async fn box_warp(_: impl Ctx, content: Table<Vector>, #[expose] rectangle: Tabl
+			let source_size = group_size;
+
+			// Safe normalization size (prevents division by zero)
+			let eps = 1e-9;
```
NIT : move this to a constant at the top of the file

Ayush2k02 commented on node-graph/nodes/vector/src/vector_nodes.rs:
```diff
@@ -791,18 +790,34 @@ async fn box_warp(_: impl Ctx, content: Table<Vector>, #[expose] rectangle: Tabl
+	let mut group_bbox: Option<[DVec2; 2]> = None;
+	for row in content.iter() {
+		if let Some(bbox) = row.element.bounding_box_with_transform(*row.transform) {
+			group_bbox = Some(match group_bbox {
```
Use this here

```
group_bbox = Some(match group_bbox {
    None => bbox,
    Some(current) => Rect::combine_bounds(current, bbox),
});
```

Like use this function which is already there in the codebase

Ayush2k02 commented on node-graph/nodes/vector/src/vector_nodes.rs:
```diff
@@ -839,13 +856,10 @@ async fn box_warp(_: impl Ctx, content: Table<Vector>, #[expose] rectangle: Tabl
 			// Transform handles in bezier curves
 			for (_, handles, _, _) in result.handles_mut() {
 				*handles = handles.apply_transformation(|pos| {
```
NIT : add these comments back , no reason for them to be removed

LilSuperUser committed:
fix(vector): make the suggested changes on group bbox warp

- move normalization epsilon to a file-level constant
- use Rect::combine_bounds for bbox merging (wrapping [DVec2; 2])
- restore blank line after dst_corners block
- restore and align inline comments across point and handle transforms
